scripts/prepare_lima.py

```diff
@@ -0,0 +1,191 @@
+"""Implementation derived from https://github.com/tloen/alpaca-lora"""
+import json
+import sys
+import os
+from pathlib import Path
+from typing import Optional
+
+import torch
+from torch.utils.data import random_split
+from tqdm import tqdm
+
+# support running without installing as a package
+wd = Path(__file__).parent.parent.resolve()
+sys.path.append(str(wd))
+
+from lit_gpt.tokenizer import Tokenizer
+
+DATA_REPO_ID = "GAIR/lima"
+DATA_FILE_NAME = "lima_data_cleaned_archive.json"
+DESTINATION_PATH = Path("data/lima")
+CHECKPOINT_DIR = Path("checkpoints/stabilityai/stablelm-base-alpha-3b")
+TEST_SPLIT_FRACTION = 0.1
+MASK_INPUTS = False  # as in alpaca-lora
+IGNORE_INDEX = -1
+SEED = 42
+
+
+def prepare(
+    destination_path: Path = DESTINATION_PATH,
+    test_split_fraction: float = TEST_SPLIT_FRACTION,
+    checkpoint_dir: Path = CHECKPOINT_DIR,
+    mask_inputs: bool = MASK_INPUTS,
+    seed: int = SEED,
+    include_multiturn_conversations: bool = False,
+    data_repo_id: str = DATA_REPO_ID,
+    ignore_index: int = IGNORE_INDEX,
+    max_seq_length: Optional[int] = None,
+    token: Optional[str] = os.getenv("HF_TOKEN")
+) -> None:
+    """Prepare the LIMA dataset for instruction tuning.
+
+    The output is a training and test dataset saved as `train.pt` and `test.pt`,
+    which stores the preprocessed and tokenized prompts and labels.
+    """
+
+    if hf_access_token is None:
```

Comment:
```suggestion
    if token is None:
```

Reply:
I renamed it to `access_token` because that's how it is also called in the UI. Might be more intuitive.

Reply:
token is the name in `scripts/download.py`. Do you want to change it there too?

Reply:
I see. If you don't mind, I suggest changing it there, too. The reason is that "token" is quite ambiguous especially in LLM contexts (e.g., think of end-of-sequence and other special tokens)

Reply:
Let's do it then